From cefbaa97d5e5579b21e6c958be69abd2eb59d70b Mon Sep 17 00:00:00 2001
From: Han Xu <han.xu@nxp.com>
Date: Tue, 31 Oct 2017 15:45:20 -0500
Subject: [PATCH 2714/5242] MLK-16745-2: arm64: dts: dedicate dtb file for QM
 lpspi

commit  81a260bb4e8302f73e77b67f1afa69ee14d87453 from
https://source.codeaurora.org/external/imx/linux-imx.git

add dedicate dtb file for lpspi nor chip on base board for i.MX8QM,
remove the CS1 pin which is not used.

 BuildInfo:
  - SCFW 9e9f6ec6, IMX-MKIMAGE 49a2866a, ATF 0
  - U-Boot 2017.03-imx_v2017.03_4.9.11_imx8_alpha+g258936c

Signed-off-by: Han Xu <han.xu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/Makefile             |    1 +
 .../dts/freescale/fsl-imx8qm-lpddr4-arm2-lpspi.dts |   54 ++++++++++++++++++++
 .../boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts  |   33 ------------
 3 files changed, 55 insertions(+), 33 deletions(-)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-lpspi.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index c696cdc..c1eae1a 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -21,6 +21,7 @@ dtb-$(CONFIG_ARCH_FSL_IMX8QM) += fsl-imx8qm-lpddr4-arm2.dtb \
 				 fsl-imx8qm-mek_ca72.dtb \
 				 fsl-imx8qm-lpddr4-arm2-it6263.dtb \
 				 fsl-imx8qm-lpddr4-arm2-it6263-dual-channel.dtb \
+				 fsl-imx8qm-lpddr4-arm2-lpspi.dtb \
 				 fsl-imx8qm-lpddr4-arm2-spdif.dtb \
 				 fsl-imx8qm-lpddr4-arm2-mqs.dtb \
 				 fsl-imx8qm-lpddr4-arm2-usb3.dtb
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-lpspi.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-lpspi.dts
new file mode 100644
index 0000000..d6d7d77
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-lpspi.dts
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8qm-lpddr4-arm2.dts"
+
+&iomuxc {
+
+	imx8qm-arm2 {
+
+		pinctrl_lpspi0: lpspi0grp {
+			fsl,pins = <
+				SC_P_SPI0_SCK_DMA_SPI0_SCK		0x0600004c
+				SC_P_SPI0_SDO_DMA_SPI0_SDO		0x0600004c
+				SC_P_SPI0_SDI_DMA_SPI0_SDI		0x0600004c
+			>;
+		};
+
+		pinctrl_lpspi0_cs: lpspi0cs {
+			fsl,pins = <
+				SC_P_SPI0_CS0_LSIO_GPIO3_IO05		0x21
+			>;
+		};
+
+	};
+};
+
+&lpspi0 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	fsl,spi-num-chipselects = <1>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_lpspi0 &pinctrl_lpspi0_cs>;
+	cs-gpios = <&gpio3 5 GPIO_ACTIVE_LOW>;
+	status = "okay";
+
+	flash: at45db041e@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "atmel,at45", "atmel,dataflash";
+		spi-max-frequency = <500000>;
+		reg = <0>;
+        };
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts
index 23041a3..bc46646 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts
@@ -402,21 +402,6 @@
 			>;
 		};
 
-		pinctrl_lpspi0: lpspi0grp {
-			fsl,pins = <
-				SC_P_SPI0_SCK_DMA_SPI0_SCK		0x0600004c
-				SC_P_SPI0_SDO_DMA_SPI0_SDO		0x0600004c
-				SC_P_SPI0_SDI_DMA_SPI0_SDI		0x0600004c
-				SC_P_SPI0_CS1_DMA_SPI0_CS1		0x0600004c
-			>;
-		};
-
-		pinctrl_lpspi0_cs: lpspi0cs {
-			fsl,pins = <
-				SC_P_SPI0_CS0_LSIO_GPIO3_IO05		0x21
-			>;
-		};
-
 		pinctrl_flexspi0: flexspi0grp {
 			fsl,pins = <
 				SC_P_QSPI0A_DATA0_LSIO_QSPI0A_DATA0	0x0600004c
@@ -719,24 +704,6 @@
 	};
 };
 
-&lpspi0 {
-	#address-cells = <1>;
-	#size-cells = <0>;
-	fsl,spi-num-chipselects = <1>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_lpspi0 &pinctrl_lpspi0_cs>;
-	cs-gpios = <&gpio3 5 GPIO_ACTIVE_LOW>;
-	status = "okay";
-
-	flash: at45db041e@0 {
-		#address-cells = <1>;
-		#size-cells = <1>;
-		compatible = "atmel,at45", "atmel,dataflash";
-		spi-max-frequency = <500000>;
-		reg = <0>;
-        };
-};
-
 &pd_dma_lpuart0 {
 	debug_console;
 };
-- 
1.7.9.5

